synapse/storage/database.py

```diff
@@ -1516,19 +1643,26 @@ def simple_search_list(self, table, term, col, retcols, desc="simple_search_list
         )
 
     @classmethod
-    def simple_search_list_txn(cls, txn, table, term, col, retcols):
+    def simple_search_list_txn(
+        cls,
+        txn: LoggingTransaction,
+        table: str,
+        term: Optional[str],
+        col: str,
+        retcols: Iterable[str],
+    ) -> Union[List[Dict[str, Any]], int]:
```

Comment:
I wonder if this should return `Optional[List[Dict[str, Any]]]` instead? This was what the old comment said, but definitely isn't what the code does. 😕

Reply:
Ah, I assume that it probably works as `0` is also falsey. Let's not try and fix it in this PR though.

Reply:
That was my assumption too. I can do a follow-up that investigates a bit more closely.